Character.cpp

```diff
@@ -98,20 +124,27 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        recalculateStatsOnLevelUp(hitPoints,
```

Comment:
You are doing the same thing with a pair of variables three times over.
write a helper function and call that helper function three times with each pair
- [ ] revise